soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -142,28 +150,26 @@ void CrowdControl::ListenToServer() {
                 continue;
             }
 
-            // If effect is a one off run, let's execute
+            // If effect is not a timed effect, execute and return result.
             if (!incomingEffect->timeRemaining) {
-                EffectResult result =
-                ExecuteEffect(incomingEffect->type.c_str(), incomingEffect->value, false);
+                EffectResult result = CrowdControl::ExecuteEffect(incomingEffect);
                 EmitMessage(tcpsock, incomingEffect->id, incomingEffect->timeRemaining, result);
             } else {
-                // check if a conflicting event is already active
+                // If another timed effect is already active that conflicts with the
+                // incoming effect, let CC retry later.
                 bool isConflictingEffectActive = false;
                 for (Effect* pack : activeEffects) {
                     if (pack != incomingEffect && pack->category == incomingEffect->category &&
                         pack->id < incomingEffect->id) {
                         isConflictingEffectActive = true;
                         EmitMessage(tcpsock, incomingEffect->id, incomingEffect->timeRemaining,
                                     EffectResult::Retry);
-
                         break;
```

Comment:
again, not something that's new to this PR, but i'm not fully understanding the logic here. when we break out of this `for (Effect* pack : activeEffects)` loop, we still make it down to L172 where `incomingEffect` is executed

if the idea that the checks within `ExecuteEffect` will also catch the incoming/active time effect conflict? if so do we need this loop here?